lib/schedule.js

```diff
@@ -0,0 +1,224 @@
+                task:operation ${sparqlEscapeUri(scheduledJob.operation)}.
+
+          ${sparqlEscapeUri(remoteDataObjectUri)} a nfo:RemoteDataObject;
+                mu:uuid ${sparqlEscapeString(remoteDataObjectId)};
+                nie:url ${sparqlEscapeString(scheduledJob.sourceUrl)};
+                adms:status ${sparqlEscapeUri(STATUS_READY_TO_BE_CACHED)};
+                rpioHttp:requestHeader ${sparqlEscapeUri(REQUEST_HEADER)};
+                dct:created ${sparqlEscapeDateTime(now)};
+                dct:modified ${sparqlEscapeDateTime(now)};
+                dct:creator ${sparqlEscapeUri(scheduledJob.uri)}.
+
+          ${sparqlEscapeUri(harvestingCollectionUri)} a hrvst:HarvestingCollection;
+                mu:uuid ${sparqlEscapeString(harvestingCollectionId)};
+                dct:creator ${sparqlEscapeUri(scheduledJob.uri)};
+                dct:hasPart ${sparqlEscapeUri(remoteDataObjectUri)}.
+
+          ${sparqlEscapeUri(dataContainerUri)} a nfo:DataContainer;
+                mu:uuid ${sparqlEscapeString(dataContainerId)};
+                task:hasHarvestingCollection ${sparqlEscapeUri(harvestingCollectionUri)}.
+
+          ${sparqlEscapeUri(taskUri)} a task:Task;
+                mu:uuid ${sparqlEscapeString(taskId)};
+                adms:status ${sparqlEscapeUri(STATUS_SCHEDULED)};
+                dct:created ${sparqlEscapeDateTime(now)};
+                dct:modified ${sparqlEscapeDateTime(now)};
+                task:operation ${sparqlEscapeUri(scheduledJob.taskOperation)};
+                task:index ${sparqlEscapeInt(0)};
+                task:inputContainer ${sparqlEscapeUri(dataContainerUri)};
+                dct:isPartOf ${sparqlEscapeUri(jobUri)}.
+        }
+      }
+    `;
+    return newJobQuery;
+  } catch(err) {
+    console.error(`[DELTA] Something went wrong constructing create-job query: ${scheduledJob.uri}`);
+    console.error(err);
+  }
+}
+
+
+
+async function addScheduledJob(scheduledJob) {
+  try {
+    const queryScheduledJobs = `
+      ${PREFIXES}
+      SELECT DISTINCT ?uri ?frequency WHERE {
+        ?uri a cogs:ScheduledJob;
+              mu:uuid ?uuid;
+              task:schedule ?schedule.
+
+  		  ?schedule schema:repeatFrequency ?frequency.
+        BIND(${sparqlEscapeUri(scheduledJob.uri)} as ?uri)
+      }
+    `;
+
+    const scheduledJobData = parseResult(await query(queryScheduledJobs));
+    addScheduledJobsToCron(scheduledJobData);
+
+  } catch(err) {
+    console.error(`[DELTA] Something went wrong while ADDING scheduled-job to cron: ${scheduledJob.uri}`);
+    console.error(err);
+  }
+}
+
+function deleteScheduledJob(scheduledJob) {
+  try {
+    const cron = CRON_JOBS[scheduledJob.uri];
+    cron.stop();
+    delete CRON_JOBS[scheduledJob.uri];
+    return;
+  } catch(err) {
+    console.error(`[DELTA] Something went wrong while DELETING scheduled-job from cron: ${scheduledJob.uri}`);
+    console.error(err);
+  }
+
+}
+
+async function updateCronFrequency(schedule) {
```

Comment:
again, I would be in favour of splitting the concerns.

the schedule.js file is more in charge of the schedling boilerplate.